src/lib/flexbox/api/style.ts

```diff
@@ -0,0 +1,171 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  Renderer,
+  OnChanges,
+  SimpleChanges,
+  KeyValueDiffers
+} from '@angular/core';
+import {NgStyle} from '@angular/common';
+
+import {BreakPointRegistry} from './../../media-query/breakpoints/break-point-registry';
+import {ResponsiveActivation} from './../responsive/responsive-activation';
+import {BaseFxDirective} from './base';
+import {MediaChange} from '../../media-query/media-change';
+import {MediaMonitor} from '../../media-query/media-monitor';
+
+/** NgStyle allowed inputs **/
+export type NgStyleType = string|string[]|Set<string>|{[klass: string]: any};
+
+/**
+ * Directive to add responsive support for ngStyle.
+ *
+ */
+@Directive({
+  selector: `
+    [style.xs],
+    [style.gt-xs],
+    [style.sm],
+    [style.gt-sm],
+    [style.md],
+    [style.gt-md],
+    [style.lg],
+    [style.gt-lg],
+    [style.xl]
+  `
+})
+export class StyleDirective extends NgStyle implements OnInit, OnChanges, OnDestroy {
```

Comment:
Need lots of tests for

*   **ClassDirective**:  `class.spec.ts`, and 
*  **StyleDirective**: `style.spec.ts`